tool/tsh/config.go

```diff
@@ -19,44 +19,64 @@ package main
 import (
 	"fmt"
 	"net"
+	"os"
+	"os/exec"
+	"runtime"
 	"strings"
 	"text/template"
 
+	"github.com/gravitational/teleport/api/constants"
 	"github.com/gravitational/teleport/api/profile"
 	"github.com/gravitational/teleport/api/utils/keypaths"
 	"github.com/gravitational/trace"
 )
 
 const sshConfigTemplate = `
-# Common flags for all {{ .clusterName }} hosts
-Host *.{{ .clusterName }} {{ .proxyHost }}
-    UserKnownHostsFile "{{ .knownHostsPath }}"
-
-# Flags for all {{ .clusterName }} hosts except the proxy
-Host *.{{ .clusterName }} !{{ .proxyHost }}
+# Common flags for all {{ .ClusterName }} hosts
+Host *.{{ .ClusterName }} {{ .ProxyHost }}
+    UserKnownHostsFile "{{ .KnownHostsPath }}"
+    IdentityFile "{{ .IdentityFilePath }}"
+    CertificateFile "{{ .CertificateFilePath }}"
+
+# Flags for all {{ .ClusterName }} hosts except the proxy
+Host *.{{ .ClusterName }} !{{ .ProxyHost }}
     Port 3022
-    {{- if .leaf }}
-    ProxyCommand ssh -p {{ .proxyPort }} {{ .proxyHost }} -s proxy:$(echo %h | cut -d '.' -f 1):%p@{{ .clusterName }}	
+    {{- if .Leaf }}
+    ProxyCommand {{ .TSHPath }} config-proxy --proxy={{ .ProxyHost }}:{{ .ProxyPort }} %h:%p "{{ .ClusterName }}"
```

Comment:
Can `ProxyCommand` be `tsh ssh %h`?
We could teach `tsh ssh` to do any string parsing to get the node name and would avoid a new subcommand.

Reply:
`tsh ssh` doesn't appear to support the subsystem (`-s`) flag, I'm not sure how much effort that would take to implement. `tsh proxy ssh` from https://github.com/gravitational/teleport/pull/7524 would do what we need, other than the string parsing.

My only concern with integrating string parsing into another command is that it will change the hostname semantics. It's not safe to always trim off everything after the first `.`, and it would be challenging to determine whether or not it should be trimmed automatically. Maybe some `--trim-host-suffix` flag could be added to `tsh proxy ssh`?

@smallinsky any thoughts on this?

Reply:
We don't have to respect the same flags as `ssh` does.
IIUC, the only difference between the current `tsh ssh` behavior and what `ProxyCommand` needs is not messing with the terminal (putting it into raw mode), and simply piping stdin/stdout.

As for hostname interpretation, we can also probe the auth API to see if a node exists.
Run the `GetNode` call with the hostname as is and with `.ClusterName` suffix trimmed and see if either exists.

Reply:
> IIUC, the only difference between the current `tsh ssh` behavior and what `ProxyCommand` needs is not messing with the terminal (putting it into raw mode), and simply piping stdin/stdout.

As I understand it, `ProxyCommand` tunnels the connection to the sshd itself, after which the local ssh client begins to communicate with the remote sshd using the regular ssh protocol. On the other hand, `tsh ssh` delivers us an already-connected channel to some remote program (e.g. bash) on the target node.

@russjones figures it might be possible to tweak `tsh ssh` to instead (optionally) open a proxied connection to just the  target sshd, but that's exactly the functionality we'll get from #7524. We could spawn a remote `sshd -i` or something, but that's getting ugly fast (and requires permissions and configuration).

The hope here is to implement this as a dumb layer on top of `ssh` for now, but then swap the implementation out to use #7524's `LocalProxy.SSHProxy()` under the hood instead of a subprocess once it's ready.

Reply:
`tsh proxy ssh` seems like a better overall solution, but it won't be ready for a while.
Let's use this PR as a stopgap and leave a `TODO` to switch to `tsh proxy ssh` later on.